### DISCUSSION TITLE
How to add body-footer containing several logos across all pages on Quarto website?

### DISCUSSION THREAD

**Discussion by michaelplynch** [quarto-dev/quarto-cli#6566] [Q&A]
### Description

I am trying to create a Quarto website  with a body footer (called footer.qmd) containing several logos as .jpg and .png files.. I am seeing two unexpected behaviours.
1. When I set body-footer as footer.qmd in_quarto.yml (as here: https://github.com/michaelplynch/test-website/blob/main/_quarto.yml) only some (it appears as every second) image in footer.qmd gets shown, as on the bottom of this page: https://michaelplynch.github.io/test-website/ .

2. For .qmd files in a sub folder (called 'People' in my MRE), footer.qmd is not rendered, instead, the title of the file 'footer.qmd' is printed in text (see here: https://michaelplynch.github.io/test-website/People/MichaelLynch.html). I can guess why this one is happening but it doesn't seem unreasonable to assume that the footer defined in _quarto.yml should apply across the website without having to copy the same file into every sub folder.

Are these bugs or am I misunderstanding how the body-footer is supposed to work? The documentation is a little light for this feature and I haven't been able to find any similar examples in the Quarto gallery to guide this.
Happy to take any feedback or insight as I dip into website design :) 

Thanks,
Michael

I have created a MRE with associated GHA here: https://github.com/michaelplynch/test-website

**Comment by mcanouil** [comment]
See <https://quarto.org/docs/websites/website-navigation.html#page-footer>.
For example, the footer on https://quarto.org:
- https://github.com/quarto-dev/quarto-web/blob/main/_quarto.yml#L30-L52

**Reply by CharlesBordet** [reply]
I have the same 2nd issue described by @michaelplynch. The `footer.qmd` defined in `_quarto.yml` only applies to files at the root directory. For files in a sub filter, the filepath is written instead of rendering the file.

The documentation on `body-footer` is sparse indeed. It is not mentioned on the links mentioned in the previous reply by @mcanouil. I could only find a reference to it here: https://quarto.org/docs/websites/website-tools.html. Note that the documentation doesn't mention we can pass a filepath there. It states we should give it "Markdown", and there is no example showing we can pass a filepath. The bug might actually be that passing a filepath makes it to be rendered on pages at the root directory. However, I do think it would be great if we can actually pass a filepath.

I could find a workaround by including the entire content of the `footer.qmd` custom file directly inside the argument of `body-footer`. But realistically it is not an elegant solution. In my case the footer code is 40 lines of code and the `_quarto.yml` file doesn't feel like the right place to put that code. It looks like this:

```yaml
website:
  body-footer: |
    ```{=html}
    <footer>
      my content
    </footer>
    ```
```

**Reply by mcanouil** [reply]
You don't need the raw block here, Quarto is adding it.
And it does support file path.

Note that `body-footer` is not the website footer (`page-footer`).

- https://github.com/quarto-dev/quarto-cli/blob/54c7493e9097db2e8dc2c50b3e4011c7258be292/src/project/types/website/website-navigation-md.ts#L704-L731

<img width="1491" height="760" alt="Screenshot of a Quarto website development environment displaying the About page with custom footer implementation.
The left panel shows the YAML configuration file with project settings including website type Demo-footer title navigation bar with Home and About links body-footer configuration HTML format options with cosmo theme brand settings custom CSS file and table of contents enabled.
The right panel displays the rendered About page with minimal content showing About this site text and the custom footer below.
The footer features a wavy decorative border at the top and a dark navy blue background with three columns.
The About section contains descriptive text explaining this is a demo Quarto website showcasing a fancy custom footer with responsive layout and modern design.
The Quick Links section provides navigation links to Home About and Quarto Docs pages.
The Connect section displays three social media icons for GitHub Bluesky and LinkedIn.
The bottom right corner shows development tools including an Add element to chat button and a green Start button along with a Quarto Preview indicator in the terminal area." src="https://github.com/user-attachments/assets/8b0395b4-9df4-4a30-9c8a-a9673d3441e4" />


An alternative:

<img width="1491" height="760" alt="Screenshot of a Quarto website development environment showing a custom footer implementation.
The left side displays the YAML configuration file with project settings including website title Demo-footer navigation bar setup and HTML formatting options with a cosmo theme and custom CSS.
The right side shows the rendered footer with a dark navy blue background featuring three columns.
The About section contains descriptive text about the demo website.
The Quick Links section provides navigation to Home About and Quarto Docs pages.
The Connect section displays social media icons for GitHub Bluesky and LinkedIn.
The footer includes copyright text mentioning 2025 Demo Footer built with Quarto in pink text." src="https://github.com/user-attachments/assets/19717e89-f13e-4ab0-89ef-f173ec799d45" />

**Reply by mcanouil** [reply]
Another alternative is to modify the `template.html` (https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/html/pandoc/template.html) from https://quarto.org/docs/journals/templates.html#html-partials

<img width="1491" height="760" alt="Screenshot of a Quarto website development environment showing the HTML template code and rendered custom footer.
The left panel displays multiple levels of code including the YAML configuration at the top with project type website title Demo-footer navigation bar with Home and About links format settings with HTML template reference theme options including cosmo and brand custom CSS file and table of contents enabled.
Below the YAML the template.html file is open showing the HTML structure including DOCTYPE declaration html tag with language attributes head section body section and the footer implementation starting at line 94.
The footer code shows a fancy-footer class container with a footer-wave div containing an SVG element for the decorative wavy border with viewBox and path definitions.
Additional footer structure includes footer-content and footer-grid divs with a footer-section containing an About heading.
The right panel shows the rendered website homepage with minimal content and the custom footer at the bottom.
The footer displays a dark navy blue background with a wavy decorative border and three columns including About section with descriptive text Quick Links section with navigation to Home About and Quarto Docs and Connect section with social media icons.
A message at the bottom right indicates Please reopen the preview." src="https://github.com/user-attachments/assets/90b9a764-5c80-4e50-9026-c8ffe46ced03" />

**Reply by mcanouil** [reply]
> I have the same 2nd issue described by @michaelplynch. The footer.qmd defined in _quarto.yml only applies to files at the root directory. For files in a sub filter, the filepath is written instead of rendering the file.

I missed that part of your comment @CharlesBordet.
You need to ensure the path starts with `/` to be absolute to the project root, otherwise it will only apply to root pages.

**Comment by CharlesBordet** [comment]
> You don't need the raw block here, Quarto is adding it.

I think that the "raw block" you mention is the `{=html}` right? If I remove this part, the text appears as is:

<img width="706" height="119" alt="image" src="https://github.com/user-attachments/assets/fd4427ff-81d5-4153-a8e0-7f7a7928ae87" />

------

When using the filepath, adding a `/` in front of it doesn't seem to produce the effect described.

Following your example, I created:

```yaml
project:
  type: website

website:
  title: "Demo-footer"
  navbar:
    left:
      - href: index.qmd
        text: Home
      - href: pages/mypage.qmd
        text: My Page
  body-footer: /_footer.html

format:
  html:
    theme:
      - cosmo
```

Which will give me:

<img width="662" height="355" alt="image" src="https://github.com/user-attachments/assets/ba075f43-f6df-4630-b327-d3fab8251d25" />

The `_footer.html` file is at the root of the project directory.

Without the `/`, it works well for `index.qmd`, but not for `pages/mypage.qmd` (same issue as described above).

Am I correct in understanding that "absolute to the project root" means having a similar mechanism as using `here::here()` in R?

---

I understand that `body-footer` is not actually the website footer. In my case I'm using it because it felt apparently easier to be able to fully customize it. 
Modifying the `template.html` is indeed the recommended way to pursue this goal.

I also notice that providing `template: template.html` works for all pages, subdirectory or not. That's an interesting workaround.

**Reply by mcanouil** [reply]
Actually, the raw block is added only when an HTML file is provided: the file content is read and enclosed in a raw HTML block, so I correct, you actually need it when writing markdown in the option.

Regarding leading `/`, if it does not work, it's likely an oversight where the option path resolution does not follow Quarto's convention on paths. I'll take a look.
Quarto's convention is `path/relative/to/document.qmd` and `/path/relative/to/root/project.qmd`, so yes, it's similar to `here`. Quarto tries to apply that convention everywhere, but there are some places that do not follow it because of how the various tools are integrated together and some oversights (some are tracked with issues).

Template and partials are the approach with the most flexibility because it allows you to get to the inside of the formats.
Note that the website/book layouts cannot be controlled as it's injected from TypeScript.

**Reply by mcanouil** [reply]
I confirm that `body-footer` does not detect `/` path as paths. (`./file.html` and `file.html` are detected as paths)

It appears `margin-header/footer` and `body-header/footer` do not follow the same code path.
`resolveProjectPaths` is used but it actually does not follow Quarto's convention so, when it sees `/file.html`, it looks for the file at the root of the file system, because it does not exist, it returns the string.
I'll dive more into this and open an issue.

**Reply by mcanouil** [reply]
The issue to track:

- #14057

**Comment by mcanouil** [comment]
For reference:

- https://github.com/quarto-dev/quarto-cli/issues/13270
- https://github.com/quarto-dev/quarto-cli/issues/1032
- https://github.com/quarto-dev/quarto-cli/issues/1948
